middleware/grpc/client_test.go

```diff
@@ -182,19 +182,26 @@ func TestStreamClientInterceptor(t *testing.T) {
 
 	client := testgrpc.NewTestServiceClient(conn)
 	span, spanCtx := tracer.StartSpanFromContext(context.Background(), "grpc.request", tracer.ResourceName("/helloworld"))
-	defer span.Finish()
 	c, err := client.StreamingOutputCall(spanCtx, &testgrpc.StreamingOutputCallRequest{})
 	require.NoError(t, err)
+	span.Finish()
 
 	c.Recv()
 
 	testTracer.Stop()
 
 	spans := testTracer.FinishedSpans()
-	require.Equal(t, 2, len(spans))
-	assert.Equal(t, strconv.Itoa(int(span.Context().TraceID())), server.ddTraceID)
+	require.Equal(t, 4, len(spans))
 	for _, finishedSpan := range spans {
 		assert.Equal(t, strconv.Itoa(int(finishedSpan.TraceID())), server.ddTraceID)
-		assert.Equal(t, strconv.Itoa(int(finishedSpan.ParentID())), server.ddParentID)
+		if finishedSpan.OperationName() == "grpc.request" {
```

Comment:
I would rename the operation-name here and on line 184 (`tracer.StartSpanFromContext`) to something more descriptive, to help make it distinguishable from the new `grpc.client` span (aka DD_SERVICE, if we used the normal tracer instead of mocktracer), and from the `grpc.message` created from receiving and sending messages on the stream.

Reply:
fixed